gandi/cli/commands/domain.py

```diff
@@ -58,8 +57,31 @@ def info(gandi, resource):
 @pass_gandi
 def create(gandi, domain, duration, owner, admin, tech, bill, background):
     """Buy a domain."""
-    result = gandi.domain.create(domain, duration, owner, admin, tech, bill,
-                                 background)
+
+    # If domain is not available, no need to continue
+    if not domain:
+            return
+
+    user_handle = gandi.call('contact.info')['handle']
+    owner_ = owner or user_handle
+    admin_ = admin or user_handle
+    tech_ = tech or user_handle
+    bill_ = bill or user_handle
+
+    msg = """
+The domain {domain} is available. If you continue, registration will be attempted with the following parameters:
+
+        Duration: {d} year(s)
+        Owner:    {o}
+        Admin:    {a}
+        Tech:     {t}
+        Billing:  {b}
+    """.format(domain=domain, d=duration, o=owner_, a=admin_, t=tech_, b=bill_)
+
+    gandi.echo(msg)
+    click.confirm("Your account will be charged. Do you want to continue?", abort=True)
```

Comment:
This is nice, we will need to add a flag to skip this prompt later, so people can use it in scripts without interaction.